workspace/services/validation.go

```diff
@@ -0,0 +1,43 @@
+package services
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+
+	"github.com/gablesiak/datatypes"
+	"gopkg.in/go-playground/validator.v9"
+)
+
+func validateData(inputStruct datatypes.InputUserData) {
```

Comment:
Please find more descriptive function names, e.g. ValidateInputStructure and ValidateRequestBody